platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/resubmitter/DefaultTransactionResubmitter.java

```diff
@@ -63,20 +66,18 @@ public List<ConsensusTransactionImpl> resubmitStaleTransactions(@NonNull final G
         }
 
         final List<ConsensusTransactionImpl> transactionsToResubmit = new ArrayList<>();
-        for (final ConsensusTransactionImpl transaction : event.getHashedData().getTransactions()) {
-            if (transaction.isSystem()) {
-                if (transaction instanceof final StateSignatureTransaction signatureTransaction) {
+        final Iterator<Transaction> iterator = event.transactionIterator();
+        while (iterator.hasNext()) {
+            final Transaction transaction = iterator.next();
+            if (transaction.getPayload().kind().equals(PayloadOneOfType.STATE_SIGNATURE_PAYLOAD)) {
```

Comment:
```suggestion
            if (Objects.equal(transaction.getPayload().kind(),PayloadOneOfType.STATE_SIGNATURE_PAYLOAD) {
```
or
```suggestion
            if (PayloadOneOfType.STATE_SIGNATURE_PAYLOAD.equals(transaction.getPayload().kind())) {
```
By doing so a NPE can never be thrown if we run in the error that kind is null. No idea if that can happen :D